### DISCUSSION TITLE
I have been toying with questions to MS Edge Copilot to see what results I get.

### DISCUSSION THREAD

**Discussion by JohnPKosh** [JohnPKosh/TestRocket#48] [Show and tell]
# Asked Copilot - write a C# minimal API that controls alexa

and this is what she replied with:
---

I can write a C# minimal API that controls alexa. Here is one possible solution:

```csharp
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Http;
using System.Net.Http;
using System.Net.Http.Json;
using System.Threading.Tasks;

var builder = WebApplication.CreateBuilder (args);
var app = builder.Build ();

// A route to send a command to alexa
app.MapPost("/command", async (HttpContext context) =>
{
    // Get the command from the request body
    var command = await context.Request.ReadFromJsonAsync<string>();

    // Check if the command is valid
    if (string.IsNullOrEmpty(command))
    {
        return Results.BadRequest("Command is required");
    }

    // Create an HTTP client
    var client = new HttpClient();

    // Set the base address and the authorization header
    client.BaseAddress = new Uri("https://api.amazonalexa.com");
    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", "your_access_token");

    // Create a JSON object with the command
    var json = new { utterance = command };

    // Post the JSON object to the alexa skill endpoint
    var response = await client.PostAsJsonAsync("/v1/alexa-skills-kit/ask", json);

    // Check if the response is successful
    if (response.IsSuccessStatusCode)
    {
        // Return a success message as a plain text response
        return Results.Text("Command sent successfully");
    }
    else
    {
        // Return an error message as a plain text response
        return Results.Text("Command failed: " + response.ReasonPhrase);
    }
});

app.Run ();
```

This code uses the `System.Net.Http` and `System.Net.Http.Json` namespaces to send HTTP requests to the alexa skill endpoint. It also uses the `Microsoft.AspNetCore.Http` namespace to create HTTP responses. The code defines a route, `/command`, that takes a command from the request body and sends it to alexa as a JSON object. The code uses the `HttpClient` class to create and send the request, and the `HttpResponseMessage` class to receive and check the response. The code uses the `Results` class to create plain text responses with success or error messages. For more information, you can check out these resources:

- [Tutorial: Create a minimal API with ASP.NET Core](^1^)
- [Create responses in Minimal API applications](^2^)
- [Alexa Skill With .NET Core](^3^)
- [Alexa.NET: An Amazon Alexa Skills SDK for .NET](^4^)
- [How to send commands to Alexa from a C# application]

Source: Conversation with Bing, 1/7/2024
(1) timheuer/alexa-skills-dotnet: An Amazon Alexa Skills SDK for .NET - GitHub. https://github.com/timheuer/alexa-skills-dotnet.
(2) Mastering Minimal APIs in .NET 8 with Examples | CodeNx - Medium. https://medium.com/codenx/minimal-apis-in-net-8-a-simplified-approach-to-build-services-eb50df56819f.
(3) Alexa Skill With .NET Core - DZone. https://dzone.com/articles/alexa-skill-withnet-core.
(4) A Minimal API with .NET 6 using C# | by Kenji Elzerman | Medium. https://medium.com/@kenslearningcurve/a-minimal-api-with-net-6-using-c-8e945913d1d.

**Comment by kc1hyc** [comment]
Ask Copilot in Edge for the date today.

You
What is the date today?
Copilot
[Today’s date is Sunday, March 10, 2024](https://www.calendardate.com/todays.htm)[1](https://www.calendardate.com/todays.htm). Here are some additional details for today:

Day Number of Year: 70
Week Number of Year: 10
Month Number of Year: 3
Leap Year: Yes
Daylight Saving Time: Yes


You
Are you sure?
Copilot
I apologize for any confusion. Let me reconfirm today’s date:

Today is Sunday, March 10, 2024. Here are some additional details for today:


You
Are we humans ready for AI?
Copilot
AI technology has already made significant strides, and its impact on various aspects of our lives will be enormous. Let’s explore some perspectives:

Eric Schmidt, former CEO and chairman of Google, warns that AI will change rapidly, and people may struggle to adapt. While improvements in AI models are accelerating, challenges remain. Issues like algorithmic bias, copyright disputes, and ethical dilemmas are yet to be fully addressed. [The risk of AI being misused for massive loss of life is a concern if not properly regulated](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/)[1](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/).
Education: Preparing the next generation for AI involves equipping them with human skills that AI cannot replicate—such as creativity, innovation, resilience, and conflict resolution. [However, we are not adequately ready for this transformation](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/)[2](https://www.ft.com/content/ab5daa64-d100-11e7-947e-f1ea5435bcc7).
Complexity: While AI has surpassed humans in specific tasks, the full range of human intelligence remains unmatched. [The intricacies of human cognition and capability are still beyond the reach of computers](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/)[3](https://www.interceptinghorizons.com/post/artificial-intelligence-is-upon-us-are-we-ready).
AI Singularity: The future of AI singularity (when AI surpasses human intelligence) is uncertain. [Safeguards are essential to prevent an AI singularity that threatens humanity](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/)[4](https://www.forbes.com/sites/forbestechcouncil/2023/11/10/are-we-ready-to-face-down-the-risk-of-ai-singularity/).
Impact: AI applications are already outpacing human abilities in various domains. [Elections, for instance, may soon be inundated with AI-generated content, posing challenges for authenticity and trust](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/)[1](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/).
In summary, while AI brings immense potential, we must navigate its risks and ensure responsible development and deployment. [Soft power will give way to innovation power, and our readiness will shape the future of AI](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/)[1](https://news.harvard.edu/gazette/story/2023/10/a-tech-warning-ai-is-coming-fast-and-its-going-to-be-rough-ride/).

You
Are you sure?
Copilot
I apologize if my previous response caused any concern. As an AI language model, I don’t hold opinions or certainty. My purpose is to provide information based on existing knowledge and perspectives. If you have further questions or need clarification, feel free to ask, and I’ll do my best to assist! 😊

Is there any existing knowledge (Truth) left,  or is there only perspective?
